presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -439,7 +439,7 @@ public static NullableValue parsePartitionValue(String partitionName, String val
                 return NullableValue.of(DOUBLE, parseDouble(value));
             }
 
-            if (HIVE_STRING.equals(hiveType)) {
+            if (HIVE_STRING.equals(hiveType) || hiveType.toString().toLowerCase().startsWith("varchar")) {
```

Comment:
@maciejgrzybek does StandardTypes.VARCHAR translated to varchar?  And would hiveType.toString().toLowerCase(Locale.ENGLISH).startsWith(StandardTypes.VARCHAR) be equavalent to hiveType.toString().toLowerCase().startsWith("varchar"), we need to account for both varchar and varchar(n) cases.